cmd/panicbot/main.go

```diff
@@ -19,8 +20,81 @@ import (
 // 	watchFile(filePath string) error
 // }
 type Config struct {
-	DiscordBotToken string
-	GuildID         string
+	DiscordBotToken  string
+	GuildID          string
+	PrimaryChannelID string
+	AlertingMethods  *AlertingMethods
+	Voting           *Voting
+}
+type AlertingMethods struct {
+	Twilio struct {
+		AccountSID        string
+		AuthToken         string
+		TwilioPhoneNumber string
+	}
+	Email struct {
+		Auth struct {
+			Indentity string
+			Username  string
+			Password  string
+			Host      string
+		}
+		From           string
+		DefaultMessage string
+	}
+}
+type Voting struct {
+	MaxBannablePerVote int
+	RequiredVotes      struct {
+		PanicAlert int
+		PanicBan   int
+	}
+	ContactOnVote struct {
+		Discord struct {
+			Users []string
+			Roles []string
+		}
+		Twilio struct {
+			PhoneNumbers []string
+		}
+		Email struct {
+			Addresses []string
+		}
+	}
+	AllowedToVote struct {
+		PanicAlert struct {
+			Users []string
+			Roles []string
+		}
+		PanicBan struct {
+			Users []string
+			Roles []string
+		}
+	}
+	VoteTimers struct {
+		PanicBanVoteTimer   int
+		PanicAlertVoteTimer int
+	}
+	Cooldown struct {
+		PanicAlert int
+		PanicBan   int
+	}
+	RateLimit struct {
+		PanicAlert struct {
+			TimeFrames struct {
```

Comment:
make this a common struct

Reply:
Still need to make this a common struct.

Pull this out from Voting struct and make RateLimit its own type. This is what is meant by common struct.

type Voting struct {
    // Code for Voting struct here
}

```
type RateLimit struct {
    PanicAlert struct {
        // Code for PanicAlert struct here
    }
    PanicBan struct {
        Day int
        Hour int
    }
}
```